crates/npmrc/src/custom_deserializer.rs

```diff
@@ -118,11 +120,29 @@ mod tests {
         env::remove_var("PACQUET_HOME");
     }
 
+    #[test]
+    fn test_default_store_dir_with_pac_env_with_prefix() {
+        env::set_var("PACQUET_HOME", "~/prefix/pacquet_home");
+        let home_dir = dirs::home_dir().unwrap();
+        let store_dir = default_store_dir();
```

Comment:
Instead of `env::set_var` (which cause global side-effect), you should pass variables to `default_store_dir` as arguments.

Reply:
I think not. this function should read variable from env, not from arguments.And after test run, env will be removed.

Reply:
Unless you can find a way to isolate env from the other tests, writing directly to and reading directly from env is more trouble than it worth down the line. I insist that you should pass them as arguments.

Reply:
But it will change some implement of this function, which is acceptable?

Reply:
Wait, I see now that `Npmrc` sets `default` to this function. You may extract the code inside the function `default_store_dir` into a function called `resolve_store_dir(...)`, then test `resolve_store_dir`.

Reply:
And now that you have created `resolve_store_dir`, you may test `resolve_store_dir` instead of `default_store_dir`.